File: Examples/SimpleElastix/SimpleTransformix.cs
using System;
using itk.simple;

namespace itk.simple.examples {
    class SimpleTransformix {
        static void Main(string[] args) {
            try {
                if (args.Length < 4) {
                    Console.WriteLine("Usage: SimpleElastix <fixedImage> <movingImage> <inputImage> <parameterFile> <output>");
                    return;
                }
                // Make transform parameter map
                SimpleElastix elastix;
                ImageFileReader reader = new ImageFileReader();
                reader.SetFileName(args[0]);
                elastix.SetFixedImage(reader.Execute());
                reader.SetFileName(args[1]);
                elastix.SetMovingImage(reader.Execute());
                elastix.SetParameterMap(ReadParameterFile(args[3]))
                elastix.Execute()

                // Instantiate transformix
                SimpleTransformix transformix

                // Read input
                reader.SetFileName(args[4])
                transformix.SetInputImage(reader.execute())
                transformix.SetParameterMapList(elastix.GetParameterMapList())

                // Perform warp
                transformix.Execute()

                // Write result image
                sitk.WriteImage(transformix.GetResultImage(), args[5])
                

            } catch (Exception ex) {
                Console.WriteLine(ex);
            }
        }
    }
}

File: Examples/SimpleElastix/SimpleTransformix.cxx
// This one header will include all SimpleITK filters and external
// objects.
#include <SimpleITK.h>
#include <iostream>
#include <stdlib.h>

namespace sitk = itk::simple;

int main ( int argc, char* argv[] ) {

  if ( argc < 4 ) {
    std::cerr << "Usage: " << argv[0] << " <fixedImage> <movingImage> <inputImage> <parameterFile> <output>\n";
    return 1;
  }

  // Make transform parameter map list
  sitk::SimpleElastix elastix;
  sitk::ImageFileReader reader;
  reader.SetFileName( std::string( argv[1] ) );
  elastix.SetFixedImage( reader.Execute() );
  reader.SetFileName( std::string( argv[2] ) );
  elastix.SetMovingImage( reader.Execute() );
  elastix.SetParameterMap( sitk.ReadParameterFile( std::string( argv[3] ) ) );
  elastix.Execute();

  // Instantiate transformix
  sitk::SimpleTransformix transformix;

  // Read input
  reader.SetFileName( std::string( argv[4] ) );
  transformix.SetInputImage( reader.Execute() );

  // Perform warp
  transformix.Execute()

  // Write result image
  sitk::ImageFileWriter writer;
  writer.SetFileName( std::string( argv[4] ) );
  writer.Execute( transformix.GetResultImage() );

  return 0;
}
